doc/source/whatsnew/v0.19.0.txt

```diff
@@ -878,6 +878,31 @@ of ``int64`` (:issue:`13988`)
    pi = pd.PeriodIndex(['2011-01', '2011-02'], freq='M')
    pi.values
 
+.. _whatsnew_0190.api.setops:
+
+Index ``+`` / ``-`` no longer used for set operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Addition and subtraction of certain Index types performed set operations
+(set union and difference). This behaviour was already deprecated since 0.15.0,
+and is now disabled and, when possible, ``+``
+and ``-`` are used for numeric operations (:issue:`8227`, :issue:`14127`).
```

Comment:
datetimelike not yet (DatetimeIndex - DatetimeIndex does still set op, but this one is a bit more work to remove). 

And `+` also works for strings, so not only numeric. Further, for numeric indices, + and - did already do numeric ops and not setops .. (to make it a bit more complex :-))

So I did not directly come up with a more accurate but simple wording, but if you have a suggestion all ears!